File: C++/palindrome.cpp
#include <iostream>
using namespace std;

int main()
{
    int n, n1, rev=0, rem;
    cout << "Enter a positive number:\n"
    << endl;
    cin >> n;
    n1=n;
    while (n != 0)
    {
        rem = n%10;
        rev = rev *10 + rem;
        n = n/10;
    }
    cout << " The reverse of the number is: " << rev << endl;
    if (n1 == n)
    {
        cout << "It is a palindrome number\n"
        << endl;
    }
    else
    {
        cout << "it is not a palindrome number"
        << endl;
    }
  
    
    return 0;
}

File: JAVA/Palindrome.java
public class Palindrome {
    public static void main(String[] args) {
        int r,sum=0,temp;
        int n = 181;

        temp = n;
        while (n>0) {
            r = n%10;
            sum = (sum*10)+r;
            n = n/10;
        }
        if (temp == sum) {
            System.out.println("This number is palindrome");
        }
        else
        System.out.println("This nuber is not palndrome");
    }
}
